File: src/Common/DataLayer/Map/IBlogPost.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using AlwaysMoveForward.AnotherBlog.Common.DomainModel;

namespace AlwaysMoveForward.AnotherBlog.Common.DataLayer.Map
{
    public interface IBlogPost
    {
        int EntryId { get; set; }
        bool IsPublished { get; set; }
        int BlogId { get; set; }
        int AuthorId { get; set; }
        string EntryText { get; set; }
        string Title { get; set; }
        DateTime DatePosted { get; set; }
        DateTime DateCreated { get; set; }
        int TimesViewed { get; set; }

        int GetCommentCount();
    }
}

File: src/Web/src/react/Models/IBlogPost.ts
import { IBlog } from './IBlog';
import { IUser } from './IUser';

export interface IBlogPost {
    Id: number;
    IsPublished: boolean;
    Blog?: IBlog;
    Author?: IUser;
    EntryText: string;
    Title: string;
    DatePosted: string;
    DateCreated: string;
    CommentCount: number;
    TimesViewed: number;
}
